### PR TITLE
[MassiveBattle] Radio with title component

### PR DESCRIPTION
**Pull request by antonia2929**
Trello card : [Adapt RewardFrom component with radio button](https://trello.com/c/tm8VfS3e/3097-adapt-rewardfrom-component-with-radio-button)

**Detailed purpose of the PR**
Add the radio button with title component

**Result and observation**
Capture d’écran 2023-06-26 à 09 38 25

**Testing Strategy**
- [ ] Already covered by tests
- [ ] Manual testing
- [ ] Unit testing

### CODE REVIEW

File: packages/@coorpacademy-components/src/atom/radio-with-title/style.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  height: 24px;
+  border-radius: 27px;
+  background: cm_grey_100;
+  border: 1px solid cm_grey_200;
+  margin-right: 16px;
+}
+
+.radio:checked {
+  appearance: none;
+  box-shadow: inset 0 0 0 0.4em cm_primary_blue;
```
Comment (jomaora) [inline packages/@coorpacademy-components/src/atom/radio-with-title/style.css:31]: 0.5em looks match more with the Figma, no?

This is with 0.4em
Capture d’écran 2023-06-26 à 11 04 16

This is with 0.5em
Capture d’écran 2023-06-26 à 11 04 05

This is the Figma
Capture d’écran 2023-06-26 à 11 04 26

**Review by jomaora — CHANGES_REQUESTED** [review]

File: packages/@coorpacademy-components/src/atom/radio-with-title/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import Title from '../title';
+import style from './style.css';
+import {RadioWithTitleProps} from './types';
+
+const RadioWithTitle = (props: RadioWithTitleProps) => {
+  const {
+    title,
+    name,
+    checked,
+    onChange = noop,
```
Comment (StefanoA1) [inline packages/@coorpacademy-components/src/atom/radio-with-title/index.tsx:13]: no onChange function provided scenario not covered
```
    onChange = noop,
```

File: packages/@coorpacademy-components/src/atom/radio-with-title/index.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    'data-name': dataName,
+    'aria-label': ariaLabel
+  } = props;
+  const [_checked, setChecked] = useState(checked);
+
+  const idRadio = uniqueId('input-radio-');
+  const handleChange = useCallback(
```
Comment (StefanoA1) [inline packages/@coorpacademy-components/src/atom/radio-with-title/index.tsx:20]: handleChange event is not covered, `input` should be triggered on a unit test to cover it

**Review by StefanoA1 — CHANGES_REQUESTED** [review]

**Review by jomaora — APPROVED** [review]

File: packages/@coorpacademy-components/src/atom/radio-with-title/test/on-change.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+test('should call handleChange when radio is checked', t => {
+  let checkedValue = false;
+  const handleChange = () => {
+    checkedValue = !checkedValue;
+    t.pass();
+  };
+  const title = {};
+  const dataName = 'input-radio';
+  const {container} = render(
```
Comment (StefanoA1) [inline packages/@coorpacademy-components/src/atom/radio-with-title/test/on-change.tsx:17]:
```suggestion
  const {container, rerender} = render(
```

File: packages/@coorpacademy-components/src/atom/radio-with-title/test/on-change.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      onChange={handleChange}
+      data-name={dataName}
+      aria-label={''}
+      name={''}
+    />
+  );
+  console.log(container);
+  const radio = container.querySelector(`[data-name="${dataName}"]`) as Element;
+  t.truthy(radio);
+  fireEvent.click(radio);
```
Comment (StefanoA1) [inline packages/@coorpacademy-components/src/atom/radio-with-title/test/on-change.tsx:30]:
```suggestion
 rerender(    <RadioWithTitle
      title={title}
      checked={checkedValue}
      onChange={handleChange}
      data-name={dataName}
      aria-label={''}
      name={''}
    />);
  fireEvent.click(radio);
```

**Pull request merged**